benchmark/src/bin/max_throughput.rs

```diff
@@ -0,0 +1,75 @@
+use benchmark::{
+    bin_utils::{
+        create_benchmarker_from_opt, linear_search_max_throughput, try_start_metrics_server,
+    },
+    ruben_opt::RubenOpt,
+    txn_generator::PairwiseTransferTxnGenerator,
+};
+use logger::{self, prelude::*};
+use structopt::StructOpt;
+
+/// During linear search, both submission rate and number of TXNs are increased.
+/// With 32 new accounts and 2 rounds, we add extra 2K TXNs as rate increases.
+const NUM_NEW_ACCOUNTS: u64 = 32;
+const NUM_REPEATED_ROUNDS: u64 = 2;
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "max_throughput",
+    author = "Libra",
+    about = "Linear Search Maximum Throughput."
+)]
+pub struct SearchOpt {
+    // Some arguments from RubenOpt will be set to linear search's default values, including
+    // num_accounts, num_rounds, num_clients, txn_pattern.
+    #[structopt(flatten)]
+    pub ruben_opt: RubenOpt,
+    /// Upper bound value of submission rate for each client.
+    #[structopt(short = "u", long = "upper_bound", default_value = "8000")]
+    pub upper_bound: u64,
+    /// Lower bound value of submission rate for each client.
+    #[structopt(short = "l", long = "lower_bound", default_value = "10")]
+    pub lower_bound: u64,
+    /// Increase step of submission rate for each client.
+    #[structopt(short = "i", long = "inc_step", default_value = "10")]
+    pub inc_step: u64,
+    /// How many times to repeat the same linear search. Each time with new accounts/TXNs.
+    #[structopt(short = "b", long = "num_searches", default_value = "10")]
+    pub num_searches: u64,
+}
+
+impl SearchOpt {
+    pub fn new_from_args() -> Self {
+        let mut args = SearchOpt::from_args();
+        args.ruben_opt.try_parse_validator_addresses();
+        args.ruben_opt.num_clients =
```

Comment:
We should have at least 1 client for 1 validator. We should also use as many cores as possible.

So in the next commit, this parameter is moved to `CommonOpt`. If user doesn't specify a value, it would be the max(#validators, #cpus).